MoreLinq/Lead.cs

```diff
@@ -65,7 +65,7 @@ public static partial class MoreEnumerable
             return _(); IEnumerable<TResult> _()
             {
                 var leadQueue = new Queue<TSource>(offset);
-                var iter = source.GetEnumerator();
+                using var iter = source.GetEnumerator();
```

Comment:
Let's revert this as its unrelated to this PR and covered by morelinq/MoreLINQ#660:

```suggestion
                var iter = source.GetEnumerator();
```